### PR TITLE
Bugfix/subtest support in step report

### PR DESCRIPTION
**Pull request by sebclrsn**
- adds subTest support in step report
- adds RemoteTest support in step report (previously not shown)
- cleans up the traceback in case of step report feature activation
- extends the documentation to show the usage of TestCase.subTest

### CODE REVIEW

File: src/pykiso/test_result/assert_step_report.py [hunk 1]
```diff
@@ -276,24 +276,23 @@ def func_wrapper(*args, **kwargs):
...
-                test_name = test_class.step_report.current_table or test_name
+                test_name = test_case_inst.step_report.current_table or test_name
 
                 # 1. Gather message, var_name, expected, received
                 # 1.1 Get message. default value: ""
-
-                if test_class.step_report.message:
-                    message = test_class.step_report.message
-                    test_class.step_report.message = ""
+                if test_case_inst.step_report.message:
```
Comment (sebastianpfischer) [inline src/pykiso/test_result/assert_step_report.py:293]: Why is codecov complaining in many of the TCs? Can we somehow improve the coverage here?

**Review by sebastianpfischer — APPROVED** [review]

**Review by sebastianpfischer — CHANGES_REQUESTED** [review]
Please improve the code coverage

File: src/pykiso/test_result/assert_step_report.py (hunk 1 above)
Comment (sebclrsn) [reply src/pykiso/test_result/assert_step_report.py:295]: I mostly changed the variable name, so I assume it wasn't tested before either
I'll see what I can do ;)

**Comment by sebastianpfischer** [comment]
How does it connect to #186?

File: src/pykiso/test_result/assert_step_report.py (hunk 1 above)
Comment (sebclrsn) [reply src/pykiso/test_result/assert_step_report.py:295]: assert_step_report coverage increased by 7%, should be fine now

**Comment by sebclrsn** [comment]
> How does it connect to #186?

Both PRs have no impact on each other, the one you linked just adds the StepReportData even if it's not activated and mine adds SubTest and RemoteTest support when the feature is activated

**Review by sebastianpfischer — APPROVED** [review]

**Pull request merged**